### DISCUSSION TITLE
Question on null value handling

### DISCUSSION THREAD

**Discussion by jaredgrabill** [ezylang/EvalEx#344] [General]
I'm curious about Null variable accessor behavior (I could not find it in the documentation.)

If accessing a variable such as 'vars[1].prop' or 'vars.sub.prop' and one of the parent references are null, what is the behavior? Null reference exception? Null cascading?

Can the behavior be customized? Are null coalesce operators like 'vars.sub?.prop' supported?

**Comment by uklimaschewski** [comment]
Hi, sorry for the delay.
At the moment, a {{NullPointerException}} is thrown when the expression is evaluated.

I think this is wrong, it should be handled like a undefined variable and throw a {{EvaluationException}}.
I'll fix that.

Null coalescing operator are no supported at the moment.

**Reply by uklimaschewski** [reply]
Fixed with #350
